regen-network/regen-web
# feat: update create/edit project forms to handle projects without credit class

## Description

Closes: #2161

For projects without credit class:
- Removes the metadata form
- At the end of the create flow, do not submit the project on chain but just set it as "published" off-chain
- Do not submit any on-chain data when editing published off chain projects but keep storing everything off-chain

Also fixes: #2177 a bug that I came across while working on this



---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...


### How to test



Go to https://deploy-preview-2178--regen-marketplace.netlify.app/ and login with a class issuer/project admin.
From https://deploy-preview-2178--regen-marketplace.netlify.app/profile/projects:
1. Create a new project with a credit class and make sure the entire flow still works as before, in particular an on chain project is created at the end of the flow
1. Create a new project without any credit class:
   - no metadata form
   - the project shouldn't be created on chain but after final submission, it should be "published" meaning that if you go to the project page before this step, it will return 404 whereas if you go to the project page after the final review step submission, it will show the project data (use the UUID in the URL to go to the project page: https://deploy-preview-2178--regen-marketplace.netlify.app/project/{someUUID})
1. Edit an on chain project, behavior should be unchanged: on chain actions gets triggered when editing anchored data (basic info, location, roles and metadata)
1. Edit a project without a credit class that hasn't been published or a project with a credit class that is not on chain yet, you should still be redirected to the project create flow
1. Edit a project without any credit class that has been published:
   - no metadata form
   - you should be redirected to the project edit flow and editing shouldn't trigger any on chain actions (contrary to 3.)

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...
